src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -404,6 +404,7 @@ void DeviceControllerFactory::Shutdown()
 {
     if (mSystemState != nullptr)
     {
+        VerifyOrDie(mSystemState->IsShutDown());
```

Comment:
should we have a comment saying why this verification exists? That way if some code path reaches this, people can understand why this check is done and `mSystemState` nullability check is insufficient (i.e. explain why delete on a system state does not implicitly mean shut down).